osu.Game/Skinning/SkinnableSound.cs

```diff
@@ -131,6 +131,14 @@ public virtual void Play()
             });
         }
 
+        protected override void LoadAsyncComplete()
+        {
+            base.LoadAsyncComplete();
+
+            if (!samplesContainer.Any())
+                updateSamples();
```

Comment:
CI (namely storyboard tests) are failing on this in particular. `DrawableStoryboardSample` is doing sample rate application using active mods in `SkinChanged()`, which is in turn called on the sample by the `base.LoadAsyncComplete()` call - but at that point the samples don't exist yet (`updateSamples()` will create them immediately after), therefore the attempt to set the sample rate will fail.

Reordering this as such fixes:

```suggestion
			// ensure samples are constructed before calling SkinChanged()
			// via base.LoadAsyncComplete().
            if (!samplesContainer.Any())
                updateSamples();
            
            base.LoadAsyncComplete();
```

but I'm not super sure this is an acceptable resolution.

Reply:
I think this is fine with the local comment for now.

Not actually sure it needs to be in `LoadAsyncComplete` in the first place, but that's for another day's investigation.